### DISCUSSION TITLE
ternary conditional operator with statement support

### DISCUSSION THREAD

**Discussion by MisterKidX** [dotnet/csharplang#7289] [Language Ideas] [answered]
in C#, the ternary conditional operator, AKA the ?: operator, is used to **assign** values based on a conditional. You can symbolize its intent as follows:

variable = conditional ? yes expression : no expression

What I would love for this syntax to offer, is statement support, in which I don't have to declare a variable and assign it the value of the ternary expression. Goes something like this:

conditional ? Foo() : Bar();

Where Foo() and Bar() can be void methods.

```
void Foo() { }
void Bar() { }

// this is allowed
bool conditional = true;
int x = conditional ? 1 : 0;

// this is not allowed, what I'm suggesting
conditional ? Foo() : Bar();

// the latter would make this obsolete
// braces added for extra drama
if (conditional)
{
    Foo();
}
else
{
    Bar();
}
```

**Comment by spydacarnage** [comment]
You're comparing the ternary to the longest form of the `if` statement. 

You could just use 
```cs
if (condition) Foo(); else Bar();
```
and then that isn't much longer, and it's a lot easier to read...

**Reply by DavidArno** [reply]
I 100% disagree with your assertion that `if (condition) Foo(); else Bar();` is easier to read than `condition ? Foo() : Bar();`

**Comment by DavidArno** [comment]
There was talk around four years ago of allowing switch statements to use the switch expression `=>` style syntax (#3038), allowing statement blocks in switch expressions (#3037) and just allowing a switch expression to be a statement expression  (#2632 and #2860), but these all seemed to have died a death as - as far as I can tell - there appears little interest amongst the language team in implementing any of them. They would typically allowed things like:
```cs
conditional switch { 
    true => Foo(), 
    false => Bar() 
};
```
Since there seems little chance of even this style of "void in expressions" happening, I think the chances of the team embracing statements in the ternary  condition operator are pretty much non-existent.

**Comment by jnm2** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/6013.

**Reply by MisterKidX** [reply]
Thanks, haven't found it when I searched

**Reply by DavidArno** [reply]
Interesting how that original discussion received lots of upvotes, whereas this time around it's just downvotes.

**Reply by jnm2** [reply]
@MisterKidX No worries. Until GitHub throws some AI at this problem, this happens all the time.

**Reply by CyrusNajmabadi** [reply]
The numbers are so small, it doesn't seem statistically significant. 

> Since there seems little chance of even this style of "void in expressions" happening

Not sure where that idea is coming from. There's a wg looking at exactly this issue.

**Comment by DavidArno** [comment]
> Not sure where that idea is coming from. There's a wg looking at exactly this issue.

Excellent news. I look forward to reading the results from that working group.
